src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient_.cs

```diff
@@ -144,39 +285,331 @@ HttpRequestMessage RequestFunc()
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            pathInImageStore.ThrowIfNull(nameof(fileChunkToUpload));
+            fileChunkToUpload.ThrowIfNull(nameof(fileChunkToUpload));
             pathInImageStore.ThrowIfNull(nameof(pathInImageStore));
             sessionId.ThrowIfNull(nameof(sessionId));
             startBytePosition.ThrowIfNull(nameof(startBytePosition));
             endBytePosition.ThrowIfNull(nameof(endBytePosition));
             length.ThrowIfNull(nameof(length));
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            requestId = requestId ?? Guid.NewGuid().ToString();
-            var url = "ImageStore/{pathInImageStore}/$/UploadChunk";
-            url = url.Replace("{pathInImageStore}", Uri.EscapeDataString(pathInImageStore.ToString()));
-            var queryParams = new List<string>();
 
-            // Append to queryParams if not null.
-            sessionId?.AddToQueryParameters(queryParams, $"session-id={sessionId}");
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.0");
-            url += "?" + string.Join("&", queryParams);
 
-            HttpRequestMessage RequestFunc()
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
```

Comment:
This is not needed for local store.

Reply:
if it's internal it shouldn't be part of the interface imo, callers will have no idea and its not documented anywhere.

Reply:
I think we can support uploading the File, but for chunk we should throw InvalidOperationException as no one will assemble these files in FileStore.

---
In reply to: [193550063](https://github.com/Microsoft/service-fabric-client-dotnet/pull/12#discussion_r193550063) [](ancestors = 193550063)